pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/kit/KitDaoImpl.java

```diff
@@ -530,30 +646,59 @@ public KitRequestShipping getKitRequestShipping(@NonNull ResultSet rs) throws SQ
+                // on insert.  Return an error to the user with some information about the existing row.
+                stmt.setString(1, kitRequestShipping.getKitLabel());
+                stmt.setString(2, kitRequestShipping.getTrackingId());
+                try (ResultSet rs = stmt.executeQuery()) {
+                    if (rs.next()) {
+                        String kitLabel = rs.getString(DBConstants.KIT_LABEL);
+                        String trackingId = rs.getString(DBConstants.TRACKING_ID);
+                        String scannedAt = rs.getString(DBConstants.DSM_SCAN_DATE);
+                        String scannedBy = rs.getString(DBConstants.SCAN_BY);
+                        dbVals.resultValue = new ScanError(kitRequestShipping.getKitLabel(),
+                                String.format("Kit %s was already associated with tracking id %s by %s at %s",
+                                        kitLabel, trackingId, scannedBy, scannedAt));
+                        return dbVals;
+                    }
+                }
+            } catch (SQLException ex) {
+                dbVals.resultValue = new ScanError(kitRequestShipping.getKitLabel(), errorMessage);
+                logger.error(errorMessage, ex);
+                return dbVals;
+            }
+
             try (PreparedStatement stmt = conn.prepareStatement(INSERT_KIT_TRACKING)) {
                 stmt.setLong(1, System.currentTimeMillis());
-                stmt.setString(2, userId);
+                stmt.setInt(2, userId);
                 stmt.setString(3, kitRequestShipping.getTrackingId());
                 stmt.setString(4, kitRequestShipping.getKitLabel());
+                stmt.setString(5, kitRequestShipping.getTrackingId());
+                stmt.setString(6, kitRequestShipping.getKitLabel());
                 int rowsAffected = stmt.executeUpdate();
                 if (rowsAffected != 1) {
-                    dbVals.resultValue = new ScanError(kitRequestShipping.getKitLabel(),
-                            String.format("Kit Label \"%s\" does not exist.\n"
-                                    + UserErrorMessages.IF_QUESTIONS_CONTACT_DEVELOPER, kitRequestShipping.getKitLabel()));
-                    logger.error(String.format("The number of affected rows for kit label %s is not 1", kitRequestShipping.getKitLabel()));
+                    dbVals.resultValue = new ScanError(kitRequestShipping.getKitLabel(), errorMessage);
                 } else {
-                    logger.info("Added tracking for kit w/ kit_label " + kitRequestShipping.getKitLabel());
+                    logger.info("Added tracking id {} for kit {}", kitRequestShipping.getTrackingId(),
+                            kitRequestShipping.getKitLabel());
+                    return dbVals;
                 }
-            } catch (Exception ex) {
-                dbVals.resultValue = new ScanError(kitRequestShipping.getKitLabel(),
```

Comment:
Rather than catching `Exception` and issuing an alert, this now catches only `SQLException` so that we can return a ScanError for some situations without sending an alert.